### DISCUSSION TITLE
Translations on polymorphic relation table

### DISCUSSION THREAD

**Discussion by sercanvirlan** [directus/directus#4152] [Feature Requests]
Hello everyone!

I'm building a mobile app and need manageable translations for all pages about 10+ pages.

For example:
Splash Page
Login Page
Home Page
Profile Page
Entity Page
Entity Detail Page

If i create all this page's as singleton object then add translations for each one, this action will create tables same count of page tables.

If there is a system table `directus_translations` with polymorphic relation, for example.

![Screen Shot 2021-02-18 at 14 40 26](https://user-images.githubusercontent.com/533209/108351987-485f5480-71f7-11eb-80f0-50bf4b45e3dc.jpg)

Pros: 
Single table, can be manageable with single  layout
Easy to query `/translations/page_id`

What are your opinions?

**Comment by benhaynes** [comment]
Not as of now, all translations are managed in O2M relationships, which means each collection gets its own translation table. While there may be other ways of handling this, our current implementation is very flexible and matches the way Directus internally handles relational data. I defer to @rijkvanzanten on the technical nuance!

**Reply by rijkvanzanten** [reply]
I concur. I like your idea @sercanvirlan as it's a very clean (1 collection) way of handling translations globally. The only downside is that it's very Directus proprietary, which we try to avoid.

That being said, it could be interesting to start treating translations as a many-to-any setup (where translations can be pointed back to any other table), as that would effectively give you this exact setup

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
